tests/test_halo_update.py

```diff
@@ -547,3 +548,88 @@ def get_horizontal_dims(dims):
     else:
         raise ValueError(f"no y dimension in {dims}")
     return y_dim, x_dim
+
+
+@pytest.mark.parametrize(
+    "layout, n_points, n_points_update, n_buffer",
+    [((1, 1), 2, "more", 0)],
+    indirect=True,
+)
+def test_halo_updater_stability(
+    zeros_quantity_list,
+    communicator_list,
+    n_points_update,
+    n_points,
+    numpy,
+    subtests,
+    boundary_dict,
+    ranks_per_tile,
+):
+    """
+    Test that that halo_updater.start()/wait() is consistent through multiple execution.
+    Test the internal buffers are re-used properly and re-cached properly.
+    """
+    BUFFER_CACHE.clear()
+    halo_updaters = []
+    for communicator, quantity in zip(communicator_list, zeros_quantity_list):
+        specification = fv3gfs.util.HaloUpdateSpec(
+            n_points,
+            quantity.data.strides,
+            quantity.data.itemsize,
+            quantity.data.shape,
+            quantity.origin,
+            quantity.extent,
+            quantity.dims,
+            quantity.np,
+            quantity.metadata.dtype,
+        )
+        halo_updater = fv3gfs.util.HaloUpdater.from_scalar_specifications(
+            comm=communicator,
+            numpy_like_module=quantity.np,
+            specifications=[specification],
+            boundaries=communicator.boundaries.values(),
+            tag=0,
```

Comment:
Later let's package this into a `communicator.get_halo_updater(specifications, other args)`, as the way of hiding `communicator.boundaries`.